pkg/aws/policies.go

```diff
@@ -1185,3 +1186,205 @@ func (c *awsClient) GetOpenIDConnectProvider(clusterID string) (string, error) {
 	}
 	return "", nil
 }
+
+func (c *awsClient) IsUpgradedNeededForRole(prefix string, accountID string) (bool, error) {
+	for _, accountRole := range AccountRoles {
+		roleName := fmt.Sprintf("%s-%s-Role", prefix, accountRole.Name)
+		role, err := c.iamClient.GetRole(&iam.GetRoleInput{
+			RoleName: aws.String(roleName),
+		})
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				case iam.ErrCodeNoSuchEntityException:
+					return false, errors.NotFound.Errorf("Roles with the prefix'%s' not found", prefix)
+				}
+			}
+			return false, err
+		}
+		isCompatible, err := c.validateRoleUpgradeVersionCompatibility(aws.StringValue(role.Role.RoleName),
+			DefaultPolicyVersion, accountID)
+
+		if err != nil {
+			return false, err
+		}
+		if isCompatible {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func (c *awsClient) UpdateTag(roleName string) error {
+	role, err := c.iamClient.GetRole(&iam.GetRoleInput{
+		RoleName: aws.String(roleName),
+	})
+	if err != nil {
+		return err
+	}
+	_, err = c.iamClient.TagRole(&iam.TagRoleInput{
+		RoleName: role.Role.RoleName,
+		Tags: []*iam.Tag{
+			{
+				Key:   aws.String(tags.OpenShiftVersion),
+				Value: aws.String(DefaultPolicyVersion),
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *awsClient) IsUpgradedNeededForOperatorRole(cluster *cmv1.Cluster, version, accountID string) (bool, error) {
+	for _, operator := range cluster.AWS().STS().OperatorIAMRoles() {
+		roleName := strings.SplitN(operator.RoleARN(), "/", 2)[1]
+		_, err := c.iamClient.GetRole(&iam.GetRoleInput{
+			RoleName: aws.String(roleName),
+		})
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				case iam.ErrCodeNoSuchEntityException:
+					return false, errors.NotFound.Errorf("Operator Role does not exists for the "+
+						"cluster '%s'", cluster.ID())
+				}
+			}
+			return false, err
+		}
+		isCompatible, err := c.ValidateClusterOperatorRolesVersion(cluster, DefaultPolicyVersion, accountID)
+		if err != nil {
+			return false, err
+		}
+		if isCompatible {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+/**
+TODO: this is done in PR 509. Need to resolve conflict when merging other PR
+*/
+
+// ValidateClusterOperatorRolesVersion ensures all operator roles for a cluster is compatible with a specific
+// openshift version
+func (c *awsClient) ValidateClusterOperatorRolesVersion(cluster *cmv1.Cluster, version string, accountID string) (
+	bool, error) {
+	for _, role := range cluster.AWS().STS().OperatorIAMRoles() {
+		isCompatible, err := c.validateOperatorRoleUpgradeVersionCompatibility(role, cluster, version, accountID)
```

Comment:
@ciaranRoche  we cannot use the same one for account role policy and operator role policy because operator role policy uses the account role prefix. So i added a new method for operator role policy. I tested a bit and seems working. I want to further test with operator role prefix and other scenarios. it should not impact the operator role policy but found this my end of day and would like more testing and clean up if needed